src/ledger/LedgerManagerImpl.cpp

```diff
@@ -940,6 +943,45 @@ LedgerManagerImpl::processFeesSeqNums(std::vector<TransactionFramePtr>& txs,
     }
 }
 
+void
+LedgerManagerImpl::prefetchTxSourceIds(std::vector<TransactionFramePtr>& txs)
+{
+    if (mApp.getConfig().PREFETCH_BATCH_SIZE > 0)
+    {
+        std::unordered_set<LedgerKey> keys;
+        for (auto const& tx : txs)
+        {
+            keys.emplace(accountKey(tx->getSourceID()));
+        }
+        auto& root = mApp.getLedgerTxnRoot();
+        root.prefetch(keys);
+    }
+}
+
+void
+LedgerManagerImpl::prefetchTransactionData(
+    std::vector<TransactionFramePtr>& txs)
+{
+    if (mApp.getConfig().PREFETCH_BATCH_SIZE > 0)
+    {
+        auto& root = mApp.getLedgerTxnRoot();
+        std::unordered_set<LedgerKey> keysToPrefetch;
+        for (auto const& tx : txs)
+        {
+            for (auto const& op : tx->getOperations())
+            {
+                if (!(tx->getSourceID() == op->getSourceID()))
+                {
+                    keysToPrefetch.emplace(accountKey(op->getSourceID()));
+                }
+                auto knownKeys = op->getLedgerKeysToPrefetch();
```

Comment:
I wonder if it would be better to change `getLedgerKeysToPrefetch` and turn it into a `insertLedgerKeysToPrefetch` that takes a reference to a set and avoid the intermediate set (causes memory churn and processing)

Reply:
Updated.